tm2/pkg/sdk/vm/keeper.go

```diff
@@ -41,16 +41,26 @@ type VMKeeper struct {
 
 	// cached, the DeliverTx persistent state.
 	gnoStore gno.Store
+
+	maxCycles int64 // max allowed cylces on VM executions
 }
 
 // NewVMKeeper returns a new VMKeeper.
-func NewVMKeeper(baseKey store.StoreKey, iavlKey store.StoreKey, acck auth.AccountKeeper, bank bank.BankKeeper, stdlibsDir string) *VMKeeper {
+func NewVMKeeper(
+	baseKey store.StoreKey,
+	iavlKey store.StoreKey,
+	acck auth.AccountKeeper,
+	bank bank.BankKeeper,
+	stdlibsDir string,
+	maxCycles int64,
```

Comment:
eventually we should pull out option fields into a VMOptions or whatever convention this sdk already uses.

Reply:
maybe worth doing now, or at least putting a comment in for the next person.

Reply:
I am adding a TODO for now. I can handle it on a different PR if it is a priority.

Reply:
By the way, I'm not perturbed (yet) by the number of parameters, and I appreciate the fact that the compiler will raise an error if we add or remove even a single parameter with the current implementation.